app/src/features/submissions/list/SubmissionsListSortMenu.tsx

```diff
@@ -1,19 +1,57 @@
-import { mdiChevronDown } from '@mdi/js';
+import { mdiChevronDown, mdiSortAscending, mdiSortDescending, mdiSortReverseVariant } from '@mdi/js';
 import Icon from '@mdi/react';
+import { MenuItem, useTheme } from '@mui/material';
 import Button from '@mui/material/Button';
 import Menu from '@mui/material/Menu';
-import Stack from '@mui/material/Stack';
 import { FuseResult } from 'fuse.js';
 import { SubmissionRecord } from 'interfaces/useSubmissionsApi.interface';
+import sortBy from 'lodash-es/sortBy';
 import React, { useState } from 'react';
-import SortMenuItem from './SortMenuItem';
+import { objectKeys } from 'utils/Utils';
 
 export type SortSubmission = FuseResult<SubmissionRecord> | SubmissionRecord;
 
+type SortProp = { key: keyof SubmissionRecord; sort: 'asc' | 'desc' };
+
 export interface ISubmissionsListSortMenuProps<TSubmission> {
+  /**
+   * Submissions to sort
+   *
+   * @type {TSubmission[]}
+   * @memberof ISubmissionsListSortMenuProps
+   */
   submissions: TSubmission[];
+
+  /**
+   * Callback fired after submissions are sorted
+   *
+   * @param {TSubmission[]} - data
+   * @returns {void}
+   * @memberof ISubmissionsListSortMenuProps
+   */
   handleSubmissions: (data: TSubmission[]) => void;
+
+  /**
+   * Sort menu items
+   * The object values will be the labels for the menu items
+   *
+   * @type {Partial<Record<keyof SubmissionRecord, string>>}
+   * @example: { name: 'Name', publish_timestamp: 'Publish Date' }
+   * @memberof ISubmissionsListSortMenuProps
+   */
   sortMenuItems: Partial<Record<keyof SubmissionRecord, string>>;
+
+  /**
+   * Manually syncs the default sort with what api responds with
+   *
+   * Note: This does not sort the submissions, only sets the default selected sort of the menu.
+   * Open to suggestions on a better way to implement this
```

Comment:
Shouldn't the Submission List maintain the sorting order, and the Sort Menu simply dictate the order it should sort them in?

Reply:
Currently the Sort Menu handles the sorting order and the sorting of the submissions. The parent component (Submissions List) handle the state for the submissions.

I did it this way to make the Sort Menu reusable accross all the submission views.